x/mm2/types/market.go

```diff
@@ -0,0 +1,116 @@
+package types
+
+import (
+	"fmt"
+)
+
+// ValidateBasic validates the market map configuration and its expected configuration.
+//
+//	In particular, this will
+//
+//	1. Ensure that the market map is valid (ValidateBasic). This ensures that each of the provider's
+//	   markets are supported by the market map.
+//	2. Ensure that each path has a corresponding ticker.
+//	3. Ensure that each path has a valid number of operations.
+//	4. Ensure that each operation has a valid ticker and that the provider supports the ticker.
+func (mm *MarketMap) ValidateBasic() error {
+	for _, market := range mm.Markets {
+		if err := market.ValidateBasic(); err != nil {
+			return err
+		}
+
+		for _, providerConfig := range market.ProviderConfigs {
+			if providerConfig.NormalizeByPair != nil {
+				if _, found := mm.Markets[providerConfig.NormalizeByPair.String()]; !found {
+					return fmt.Errorf("provider's (%s) pair for normalization (%s) was not found in the marketmap", providerConfig.Name, providerConfig.NormalizeByPair.String())
```

Comment:
Needs test.

Reply:
tested by "provider config includes a ticker that is not supported"